### PR TITLE
Preserve data

### PR DESCRIPTION
**Pull request by cancelei**
#### Changes made : 
 - Added functions for written and reading files
 - Added Added function to recreate objects on the app starts
 #### Features : 
- Create and save data into files
- Read data from the files

### CODE REVIEW

File: lib/app.rb [hunk 1]
```diff
@@ -76,7 +75,8 @@ def create_rental
     puts 'Rental created successfully!'
   end
 
-  def list_rentals_for_person(person_id)
+  def list_rentals_for_person
+    retrieve_id
     rentals = @rentals.select { |rental| rental.person.id == person_id }
 
     if rentals.empty?
```
Comment (ShadyShawkat) [inline lib/app.rb]: - [ ] Kindly fix the following error that appeared when I tried to show a rental.

File: lib/app.rb [hunk 2]
```diff
@@ -3,18 +3,17 @@
...
-require_relative 'modules/mod_people'
-require_relative 'modules/mod_books'
-require_relative 'modules/mod_rentals'
 
+require 'json'
+require 'fileutils'
 class App
   include PeopleMod
   include BooksMod
   include RentalsMod
```
Comment (ShadyShawkat) [inline lib/app.rb:15]: -  [ ] Kindly make sure that the app is working properly. This error appeared when I tried to run the app. You can fix this easily by requiring the modules called here.
image

**Review by ShadyShawkat — CHANGES_REQUESTED** [review]
## Hi @cancelei, 

Good job so far! 👍🏻 
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight
- Correct use of GitFlow ✔️ 
- Descriptive commit messages ✔️ 
- Professional looking pull request ✔️ 

Image

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by shahadat3669 — APPROVED** [review]
Hi @cancelei , 

Your project is complete!  There is nothing else to say other than it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
✅ On exit from the student's app  the data is being saved
✅ On startup of their app load all data from files
✅ The code is top-notch! 💯

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

## Have any doubt ❓

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

<table  width="100%" align="center">
  <tr>
    <td colspan="4" align="center">
      <strong>Contact me: Shahadat Hossain</strong>
    </td>
  </tr>
  <tr>
    <td align="center">
      <a href="https://github.com/shahadata3669">
        GitHub
      </a>
    </td>
    <td align="center">
      <a href="https://www.linkedin.com/in/shahadat-cseng/">
        LinkedIn
      </a>
    </td>
    <td align="center">
      <a href="https://app.slack.com/client/T47CT8XPG/D05GT4E4WAY/rimeto_profile/U04426SD71V">
        Slack
      </a>
    </td>
    <td align="center">
      <a href="mailto:shahadat3669@gmail.com?subject=Microverse%20Code%20Review">
        Email
      </a>
    </td>
  </tr>
</table>

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**